File: src/euler/exercise10.go
/*
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
*/

package euler

import (
    "euler/primes"
    "euler/utils"
)

func Exercise10() int64 {
    p := primes.Eratosthenes(2e6)
    total := utils.AccumulateInt64(p)
    return total
}

File: cpp/src/exercise10.cpp
#include <iostream>

#include "euler-tools.h"

uint64_t Euler::exercise10() {
    std::vector<bool> primes = eratosthenes(2000000);
    uint64_t total = 0;
    for(std::vector<bool>::size_type i = 2; i < primes.size(); ++i) {
        if(primes[i]) {
            total += i;
        }
    }
    std::cout << std::endl;
    return total;
}
